packages/opentelemetry-core/src/platform/node/hex-to-bytes.ts

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * converts id string into bytes
+ * @param hexStr - id of span
+ */
+export function hexToBytes(hexStr: string): Uint8Array {
+  const hexStrLen = hexStr.length;
+  let hexAsciiCharsStr = '';
+  for (let i = 0; i < hexStrLen; i += 2) {
+    const hexPair = hexStr.substring(i, i + 2);
+    const hexVal = parseInt(hexPair, 16);
+    hexAsciiCharsStr += String.fromCharCode(hexVal);
+  }
+
+  return Buffer.from(hexAsciiCharsStr, 'ascii');
```

Comment:
Why not `Buffer.from(hexStr, 'hex')` and remove the translation from hex to ascii?
There are differences for malformed ids (e.g. `len % 2 !== 0` or `'zz'`) but my feeling is that `Buffer.from(hexStr, 'hex')` is "more correct"

Reply:
This function is removed from the PR.